src/rust/engine/fs/store/src/snapshot.rs

```diff
@@ -276,6 +276,36 @@ impl Snapshot {
       .await
     }
   }
+
+  /// # Safety
+  ///
+  /// This should only be used for testing, as this allows creating an invalid Snapshot.
+  pub unsafe fn create_for_testing_ffi(
```

Comment:
Does this need to be `unsafe`?

Reply:
Not technically, but it very unsafe if our production code ever used this. While the function name already makes that clear, I wanted to leverage Rust's mechanism to be extra defensive about this.

Reply:
My concern is that, when I read `unsafe` in Rust code, I am primed to look for memory unsafety. Why not just tag it `#[test]` so it cannot be compiled into the production binary?

Reply:
Because the test happens in Python land, it _does_ need to be included in our production binary / native extension. Only, both Python and Rust production code should be careful to not use this function.